src/catalog/CatalogPage.tsx

```diff
@@ -70,25 +93,35 @@ const CatalogPage = () => {
     );
   }
 
-  const totalCourses = courseData?.results?.length ?? 0;
-  const pageCount = Math.ceil((courseData?.total || totalCourses) / DEFAULT_PAGE_SIZE);
+  const totalCourses = displayData?.results?.length ?? 0;
+  const pageCount = Math.ceil((displayData?.total || totalCourses) / DEFAULT_PAGE_SIZE);
 
   return (
     <Container fluid={false} size="xl" className="pt-5.5 mb-6">
       <SubHeader
-        title={intl.formatMessage(messages.exploreCourses)}
+        title={getPageTitle({
+          intl,
+          lastSearchQuery,
+          searchString,
+          courseData,
+        })}
         className={classNames({ 'mx-2.5': isMedium })}
       />
       {totalCourses > 0 ? (
         <>
-          <SearchField
-            key="search-field"
-            className={classNames({
-              'w-auto mx-2.5 mb-0': isMedium,
-              'mb-4 w-25': !isMedium,
-            })}
-            placeholder={intl.formatMessage(messages.searchPlaceholder)}
-          />
+          {getConfig().ENABLE_COURSE_DISCOVERY && (
+            <SearchField
+              key="search-field"
+              className={classNames({
+                'w-auto mx-2.5 mb-0': isMedium,
+                'mb-4 w-25': !isMedium,
+              })}
+              placeholder={intl.formatMessage(messages.searchPlaceholder)}
+              value={searchString}
+              onSubmit={handleSearch}
+              onClear={handleClearSearch}
+            />
+          )}
           <DataTable
             isLoading={isFetching}
             showFiltersInSidebar={!isMedium}
```

Comment:
[inform]: While testing on mobile screens, I identified an issue with the DataTable component. I have created a separate issue for this here: https://github.com/openedx/paragon/issues/3940

Reply:
https://github.com/openedx/paragon/pull/3986 has landed, it'd be great to get a Paragon bump on this MFE so we can use `numBreakoutFilters={0}` here